isis/src/base/objs/Spice/Spice.cpp

```diff
@@ -170,41 +170,54 @@ namespace Isis {
     //  missions (e.g., MESSENGER) may augment the s/c SPK with new planet
     //  ephemerides. (2008-02-27 (KJB))
     if (m_usingNaif) {
-      std::string aleIsdStr = ale::load(cube.fileName().toStdString(), "", "isis");
-      
-      isd = json::parse(aleIsdStr);  
-      json aleNaifKeywords = isd["NaifKeywords"]; 
-      m_naifKeywords = new PvlObject("NaifKeywords", aleNaifKeywords); 
-
-      if (noTables) {
-        load(kernels["TargetPosition"], noTables);
-        load(kernels["InstrumentPosition"], noTables);
-        load(kernels["InstrumentPointing"], noTables);
-      }
-
-      if (kernels.hasKeyword("Frame")) {
-        load(kernels["Frame"], noTables);
-      }
+      try { 
+        // try using ALE
+        std::ostringstream kernel_pvl;
+        kernel_pvl << kernels;
+
+        json props;
+        props["kernels"] = kernel_pvl.str();
+
+        isd = json::parse(ale::loads(cube.fileName().toStdString(), props.dump(), "isis"));
+        json aleNaifKeywords = isd["NaifKeywords"];
+        m_naifKeywords = new PvlObject("NaifKeywords", aleNaifKeywords);
+        
+        // Still need to load clock kernels for now 
```

Comment:
Any reason why this is the case?

Reply:
Has to do with the weird way times are added to naif keywords.

in Spice.cpp:
```
 1006   iTime Spice::getClockTime(QString clockValue, int sclkCode, bool clockTicks) {
 1007     if (sclkCode == -1) {
 1008       sclkCode = naifSclkCode();
 1009     }
 1010
 1011     iTime result;
 1012
 1013     QString key = "CLOCK_ET_" + Isis::toString(sclkCode) + "_" + clockValue;
 1014     QVariant storedClockTime = getStoredResult(key, SpiceDoubleType);
```

Ale isn't shipping the information to construct this string. The only way it seems to add this functionality without changing all Camera Plugins is to pack this info into the ISD.